rocky/rocky/scheduler.py

```diff
@@ -262,14 +277,41 @@ def health(self) -> ServiceHealth:
         health_endpoint.raise_for_status()
         return ServiceHealth.model_validate_json(health_endpoint.content)
 
-    def get_task_stats(self, organization_code: str, task_type: str) -> dict:
+    def _get(self, path: str) -> dict:
+        """Helper to do a get request."""
         try:
-            res = self._client.get(f"/tasks/stats/{task_type}-{organization_code}")
+            res = self._client.get(path)
             res.raise_for_status()
         except HTTPError:
-            raise SchedulerError()
-        task_stats = json.loads(res.content)
-        return task_stats
+            raise SchedulerError(path)
+        return res.json()
+
+    def _get_task_stats(self, scheduler_id: str) -> dict:
+        """Return task stats for specific scheduler."""
+        return self._get(f"/tasks/stats/{scheduler_id}")
+
+    def get_task_stats(self, organization_code: str, task_type: str) -> dict:
+        """Return task stats for specific organization and task combination."""
+        return self._get_task_stats(scheduler_id=f"{task_type}-{organization_code}")
+
+    @staticmethod
+    def _merge_stat_dicts(dicts: list[dict]) -> dict:
+        """Merge multiple stats dicts."""
+        stat_sum: dict[str, collections.Counter] = collections.defaultdict(collections.Counter)
+        for dct in dicts:
+            for timeslot, counts in dct.items():
+                stat_sum[timeslot].update(counts)
+        return dict(stat_sum)
+
+    def get_all_task_stats(self, task_type: str, scheduler_ids: list | None = None) -> dict:
+        """Return all task stats for specific task type (e.g. 'boefje')."""
+        return SchedulerClient._merge_stat_dicts(
+            dicts=[
+                self._get_task_stats(scheduler_id=scheduler.get("id"))
+                for scheduler in self._get(f"/schedulers/by_type/{task_type}")
+                if (scheduler_ids is None or scheduler["id"] in scheduler_ids)
+            ]
+        )
```

Comment:
This would be a nice addition to implement this in the scheduler to be able to get stats for multiple scheduler id's. That way we could off-load rocky doing the filtering, by letting the scheduler filter this in db-time.

Reply:
I agree filtering these jobs in Rocky should not be done, they should be filtered in The scheduler.
the complexity is that we have no easy API on the scheduler to filter on tasks for a list of schedulers, correct?

Reply:
I simplified this function. I would say that extending the scheduler to create stats for multiple schedulers is a nice feature request but out of scope for a MVP (unless it turns out later that python in rocky is too slow to merge a couple of dictionaries)